### PR TITLE
add two commands for TAP style frontend testing

### PR DESCRIPTION
**Pull request by Raynos**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -158,11 +159,49 @@ async function handleMessage (data) {
...
+
+    if (typeof firstArg !== 'string') {
+      // nothing can be done here
+      return {}
+    }
+
+    let exitCode = -1
+    if (firstArg.indexOf('# ok') === 0) {
+      exitCode = 0
+    } else if (firstArg.indexOf('# fail ') === 0) {
```
Comment (jwerle) [inline index.js:183]: where does this output come from?

Comment (Raynos) [reply index.js:187]: There is a second PR coming

File: index.js [hunk 2]
```diff
@@ -158,11 +159,53 @@ async function handleMessage (data) {
...
+
+    console.error('Got an uncaught in test', opts)
+
+    process.nextTick(() => {
+      throw new Error('FRONTEND TEST UNCAUGHT: ' + opts.err.message)
+    })
+  } else if (value?.method === 'testConsole') {
+    const opts = value.arguments[0]
+
+    const args = JSON.parse(opts.args)
```
Comment (chicoxyzzy) [inline index.js:174]: this may throw. should we wrap it in try-catch?

File: index.js [hunk 3]
```diff
@@ -158,11 +159,53 @@ async function handleMessage (data) {
...
+    const firstArg = args[0]
+
+    console.log.apply(console, args)
```
Comment (chicoxyzzy) [inline index.js:177]: `console.log(...args)` is a bit shorter :)

**Comment by Raynos** [comment]
Related to https://github.com/socketsupply/op-examples/pull/2

File: index.js (hunk 2 above)
Comment (Raynos) [reply index.js:174]: normally yes, but it's part of `testConsole` so its ok to have it throw

File: index.js (hunk 3 above)
Comment (Raynos) [reply index.js:177]: willfix

**Review by jwerle — APPROVED** [review]

**Pull request merged**